streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/AssignorConfiguration.java

```diff
@@ -153,7 +153,8 @@ public RebalanceProtocol rebalanceProtocol() {
                     throw new IllegalArgumentException("Unknown configuration value for parameter 'upgrade.from': " + upgradeFrom);
             }
         }
-        return RebalanceProtocol.EAGER;
+
+        return RebalanceProtocol.COOPERATIVE;
```

Comment:
Hey @ableegoldman , I think you need to add the missing "upgrade from" versions to the `configuredMetadataVersion` method below as well. Otherwise, creating Streams will just throw an exception when you pass (for example) `UPGRCE_FROM_23`.

Reply:
Oops, thanks yeah good catch! That reminds me I still need to add test that actually configures the assignor using these.

Reply:
Also, not sure if it makes sense to have a test that you can actually construct and start Streams with all versions of "upgrade.from", which would have caught this.

Reply:
Well no, I just added a test that uses `UPGRADE_FROM_23` (and one that doesn't)

Reply:
Which would have caught that. I'm wondering if it actually makes more sense to not throw an exception in the `default` and just check beforehand whether its a valid config (ie is it contained in some "valid config values" set, which we actually may already have somewhere in `StreamsConfig`?)
It's not really practical to have to add new `UPGRADE_FROM` versions to an increasing number of places in the code or else get an exception thrown. Some cleanup to think about once this is over

Reply:
Agreed on the cleanup front. The current code makes sense when you just have one usage. As soon as we add a second, we probably need a different strategy.